parquet-variant/tests/integration_test.rs

```diff
@@ -0,0 +1,394 @@
+
+        // Ensure the JSON can be parsed back
+        let parsed: Value = serde_json::from_str(&json_string).unwrap();
+
+        // Ensure our direct Value conversion matches
+        let direct_value = variant_to_json_value(&variant).unwrap();
+        assert_eq!(parsed, direct_value, "Mismatch for variant: {:?}", variant);
+    }
+}
+
+#[test]
+fn test_string_escaping_edge_cases() {
+    // Test various escape sequences
+    let escaped_string = Variant::String("line1\nline2\ttab\"quote\"\\backslash");
+    let expected_json = "\"line1\\nline2\\ttab\\\"quote\\\"\\\\backslash\"";
+    assert_eq!(
+        variant_to_json_string(&escaped_string).unwrap(),
+        expected_json
+    );
+
+    // Test Unicode characters
+    let unicode_string = Variant::String("Hello 世界 🌍");
+    let json_result = variant_to_json_string(&unicode_string).unwrap();
+    assert!(json_result.contains("Hello 世界 🌍"));
+    assert!(json_result.starts_with('"') && json_result.ends_with('"'));
+}
+
+#[test]
+fn test_json_roundtrip_compatibility() {
+    // Test that our JSON output can be parsed back by serde_json
+    let test_cases = vec![
+        Variant::Null,
+        Variant::BooleanTrue,
+        Variant::BooleanFalse,
+        Variant::Int8(0),
+        Variant::Int8(127),
+        Variant::Int8(-128),
+        Variant::String(""),
+        Variant::String("simple string"),
+        Variant::String("string with\nnewlines\tand\ttabs"),
+        Variant::ShortString("short"),
+    ];
+
+    for variant in test_cases {
+        let json_string = variant_to_json_string(&variant).unwrap();
+
+        // Ensure the JSON can be parsed back
+        let parsed: Value = serde_json::from_str(&json_string).unwrap();
+
+        // Ensure our direct Value conversion matches
+        let direct_value = variant_to_json_value(&variant).unwrap();
+        assert_eq!(parsed, direct_value, "Mismatch for variant: {:?}", variant);
+    }
+}
+
+#[test]
+fn test_buffer_writing() {
+    use parquet_variant::variant_to_json;
+    use std::io::Write;
+
+    let variant = Variant::String("test buffer writing");
+
+    // Test writing to a Vec<u8>
+    let mut buffer = Vec::new();
+    variant_to_json(&mut buffer, &variant).unwrap();
+    let result = String::from_utf8(buffer).unwrap();
+    assert_eq!(result, "\"test buffer writing\"");
+
+    // Test writing to a custom writer
+    struct CustomWriter {
+        data: Vec<u8>,
+    }
+
+    impl Write for CustomWriter {
+        fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+            self.data.extend_from_slice(buf);
+            Ok(buf.len())
+        }
+
+        fn flush(&mut self) -> std::io::Result<()> {
+            Ok(())
+        }
+    }
+
+    let mut custom_writer = CustomWriter { data: Vec::new() };
+    variant_to_json(&mut custom_writer, &variant).unwrap();
+    let result = String::from_utf8(custom_writer.data).unwrap();
+    assert_eq!(result, "\"test buffer writing\"");
+}
```

Comment:
Since `Vec` already implements `Write` I think you can simplify test to something like

```suggestion
fn test_buffer_writing() {
    use parquet_variant::variant_to_json;
    use std::io::Write;

    let variant = Variant::String("test buffer writing");

    // Test writing to a Vec<u8>
    let mut buffer = Vec::new();
    variant_to_json(&mut buffer, &variant).unwrap();
    let result = String::from_utf8(buffer).unwrap();
    assert_eq!(result, "\"test buffer writing\"");

    let mut buffer = vec![];
    variant_to_json(&mut buffer, &variant).unwrap();
    let result = String::from_utf8(buffer).unwrap();
    assert_eq!(result, "\"test buffer writing\"");
}
```